### PR TITLE
Add tests for API endpoint

### PR DESCRIPTION
**Pull request by em411**
| Q               | A
| --------------- | -----
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | ---
| License         | MIT

### CODE REVIEW

File: tests/Behat/Context/Api/BlockContext.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+{
+    private $apiClient;
+
+    private $responseChecker;
+
+    public function __construct(
+        ApiClientInterface $apiClient,
+        ResponseCheckerInterface $responseChecker
+    ) {
+        $this->apiClient       = $apiClient;
```
Comment (hmfilar) [inline tests/Behat/Context/Api/BlockContext.php:29]: We don't use assignments alignment (you can configure phpStorm to not use it when reformatting, Settings -> Editor/Code Style -> uncheck "align consecutive assignments")

File: tests/Behat/Context/Api/BlockContext.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
+<?php
+/*
```
Comment (hmfilar) [inline tests/Behat/Context/Api/BlockContext.php:2]: according to the BitBagBible there should be one blank line here

**Review by hmfilar — COMMENTED** [review]
I think you should get rid of assignments alignment (you can configure phpStorm to not use it when reformatting, Settings -> Editor/Code Style -> uncheck "align consecutive assignments") in your constructors. You should also add a blank line between `<?php` and the "This file has been..." comment.
Also we should use xml for service configuration instead of yaml, although I'm not sure what you're supposed to do here, as there already are lots of services configured in yaml in this plugin.

File: tests/Behat/Context/Api/BlockContext.php (hunk 1 above)
Comment (em411) [reply tests/Behat/Context/Api/BlockContext.php:29]: I think, that was mentioned in BitBag Bible -> Code Style -> 10. Good to follow practices

File: tests/Application/config/sylius/1.9/packages/security.yaml [hunk 3]
```diff
@@ -45,7 +40,7 @@ security:
                 csrf_token_id: admin_authenticate
             remember_me:
                 secret: "%env(APP_SECRET)%"
-                path: "/%sylius_admin.path_name%"
+                path: /admin
```
[the remark this answers is not in the document]
Comment (em411) [reply tests/Application/config/sylius/1.9/packages/security.yaml:43]: It's an exact copy of Sylius 1.10 security.yaml, but you are probably right, I'll change it

**Review by hmfilar — APPROVED** [review]

File: tests/Application/config/sylius/1.9/packages/security.yaml [hunk 4]
```diff
@@ -134,15 +129,10 @@ security:
...
 
         - { path: "%sylius.security.shop_regex%/register", role: IS_AUTHENTICATED_ANONYMOUSLY }
         - { path: "%sylius.security.shop_regex%/verify", role: IS_AUTHENTICATED_ANONYMOUSLY }
 
         - { path: "%sylius.security.admin_regex%", role: ROLE_ADMINISTRATION_ACCESS }
-        - { path: "%sylius.security.api_regex%/.*", role: ROLE_API_ACCESS }
         - { path: "%sylius.security.shop_regex%/account", role: ROLE_USER }
-
-        - { path: "%sylius.security.new_api_admin_regex%/.*", role: ROLE_API_ACCESS }
-        - { path: "%sylius.security.new_api_shop_regex%/.*", role: IS_AUTHENTICATED_ANONYMOUSLY }
```
Comment (patrick477) [inline tests/Application/config/sylius/1.9/packages/security.yaml:148]: are you sure it should be removed?

File: tests/Behat/Context/Api/BlockContext.php [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            $this->apiClient->getLastResponse()),
+            $count
+        );
+    }
+
+    /**
+     * @Then I should see block with code :block
+     * @Given I view block with code :block
+     */
+    public function iShouldSeeBlockWithCode($code): void
```
Comment (patrick477) [inline tests/Behat/Context/Api/BlockContext.php:57]: string $code?

File: tests/Behat/Context/Api/BlockContext.php [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+     */
+    public function iWantToBrowseBlocks(): void
+    {
+        $this->apiClient->index();
+    }
+
+    /**
+     * @Then /^I should see (\d+) blocks in the list$/
+     */
+    public function iShouldSeeBlocksInTheList($count): void
```
Comment (patrick477) [inline tests/Behat/Context/Api/BlockContext.php:45]: int $count

File: tests/Behat/Context/Api/FrequentlyAskedQuestionContext.php [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+     */
+    public function iWantToBrowseFAQs(): void
+    {
+        $this->apiClient->index();
+    }
+
+    /**
+     * @Then /^I should see (\d+) question(?:s)? in the list$/
+     */
+    public function iShouldSeeQuestionsInTheList($count): void
```
Comment (patrick477) [inline tests/Behat/Context/Api/FrequentlyAskedQuestionContext.php:46]: int $count

File: tests/Behat/Context/Api/FrequentlyAskedQuestionContext.php [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+        Assert::count($this->responseChecker->getCollection(
+            $this->apiClient->getLastResponse()),
+            $count
+        );
+    }
+
+    /**
+     * @Given /^I should see the "([^"]*)" question$/
+     */
+    public function iShouldSeeTheQuestion($code): void
```
Comment (patrick477) [inline tests/Behat/Context/Api/FrequentlyAskedQuestionContext.php:57]: string $code

File: tests/Behat/Context/Api/MediaContext.php [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+     */
+    public function iWantToBrowseMedia(): void
+    {
+        $this->apiClient->index();
+    }
+
+    /**
+     * @Then /^I should see (\d+) media in the list$/
+     */
+    public function iShouldSeeMediaInTheList($count): void
```
Comment (patrick477) [inline tests/Behat/Context/Api/MediaContext.php:46]: int $count

File: tests/Behat/Context/Api/PageContext.php [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+        Assert::true(
+            $this->responseChecker->hasItemWithTranslation(
+                $this->apiClient->getLastResponse(),
+                'en_US',
+                'content',
+                $content
+            )
+        );
+    }
+}
```
Comment (patrick477) [inline tests/Behat/Context/Api/PageContext.php:107]: typing is missing in methods

File: tests/Behat/Context/Api/SectionContext.php [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+        Assert::false(
+            $this->responseChecker->hasTranslation(
+                $this->apiClient->getLastResponse(),
+                "en_US",
+                "name",
+                "That shouldn't exist"
+            ), "Section has missing name"
+        );
+    }
+}
```
Comment (patrick477) [inline tests/Behat/Context/Api/SectionContext.php:77]: typing is missing in methods

File: tests/Behat/Context/Transform/MediaContext.php [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+    /** @var MediaRepositoryInterface */
+    private $mediaRepositoryInterface;
+
+    /** @var string */
+    private $locale;
+
+    public function __construct(MediaRepositoryInterface $mediaRepositoryInterface, string $locale = 'en_US')
+    {
+        $this->mediaRepositoryInterface = $mediaRepositoryInterface;
+        $this->locale                   = $locale;
```
Comment (patrick477) [inline tests/Behat/Context/Transform/MediaContext.php:31]: format

File: tests/Behat/Context/Transform/PageContext.php [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+    /** @var PageRepositoryInterface */
+    private $pageRepository;
+
+    /** @var string */
+    private $locale;
+
+    public function __construct(PageRepositoryInterface $pageRepository, string $locale = 'en_US')
+    {
+        $this->pageRepository = $pageRepository;
+        $this->locale         = $locale;
```
Comment (patrick477) [inline tests/Behat/Context/Transform/PageContext.php:31]: format

**Review by leszczuu — APPROVED** [review]

**Pull request merged**